Help - Opam failing to build

Hey folks, I am trying to run an app that was made with Opam.

I am on Ubuntu 20.10, 

Current Opam version is 2.0.5
GCC version (Ubuntu 10.2.0-13ubuntu1) 10.2.0

Does anyone know how to fix this?
I have no idea how this works to at least try something.

```
<><> Updating repositories ><><><><><><><><><><><><><><><><><><><><><><><><><><>
[default] no changes from https://opam.ocaml.org
➜  plotkicadsch git:(master)    opam switch create 4.09.0 --jobs=1                


<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ocaml-base-compiler.4.09.0] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] The compilation of ocaml-base-compiler failed at "/home/lheck/.opam/opam-init/hooks/sandbox.sh build make -j1 world".
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base

#=== ERROR while compiling ocaml-base-compiler.4.09.0 =========================#
# context     2.0.5 | linux/x86_64 |  | https://opam.ocaml.org#a2215354
# path        ~/.opam/4.09.0/.opam-switch/build/ocaml-base-compiler.4.09.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build make -j1 world
# exit-code   2
# env-file    ~/.opam/log/ocaml-base-compiler-162351-c61e7b.env
# output-file ~/.opam/log/ocaml-base-compiler-162351-c61e7b.out
### output ###
# [...]
# gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -fno-tree-vrp -g -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -DOCAML_STDLIB_DIR='"/home/lheck/.opam/4.09.0/lib/ocaml"'  -o spacetime_byt_bpic.o spacetime_byt.c
# gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -fno-tree-vrp -g -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -DOCAML_STDLIB_DIR='"/home/lheck/.opam/4.09.0/lib/ocaml"'  -o afl_bpic.o afl.c
# gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -fno-tree-vrp -g -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -DOCAML_STDLIB_DIR='"/home/lheck/.opam/4.09.0/lib/ocaml"'  -o unix_bpic.o unix.c
# gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -fno-tree-vrp -g -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -DOCAML_STDLIB_DIR='"/home/lheck/.opam/4.09.0/lib/ocaml"'  -o bigarray_bpic.o bigarray.c
# gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -fno-tree-vrp -g -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE  -DOCAML_STDLIB_DIR='"/home/lheck/.opam/4.09.0/lib/ocaml"'  -o main_bpic.o main.c
# rm -f libcamlrun_pic.a && ar rc libcamlrun_pic.a  interp_bpic.o misc_bpic.o stacks_bpic.o fix_code_bpic.o startup_aux_bpic.o startup_byt_bpic.o freelist_bpic.o major_gc_bpic.o minor_gc_bpic.o memory_bpic.o alloc_bpic.o roots_byt_bpic.o globroots_bpic.o fail_byt_bpic.o signals_bpic.o signals_byt_bpic.o printexc_bpic.o backtrace_byt_bpic.o backtrace_bpic.o compare_bpic.o ints_bpic.o floats_bpic[...]
# gcc -shared -o libcamlrun_shared.so interp_bpic.o misc_bpic.o stacks_bpic.o fix_code_bpic.o startup_aux_bpic.o startup_byt_bpic.o freelist_bpic.o major_gc_bpic.o minor_gc_bpic.o memory_bpic.o alloc_bpic.o roots_byt_bpic.o globroots_bpic.o fail_byt_bpic.o signals_bpic.o signals_byt_bpic.o printexc_bpic.o backtrace_byt_bpic.o backtrace_bpic.o compare_bpic.o ints_bpic.o floats_bpic.o str_bpic.o [...]
# /usr/bin/ld: backtrace_bpic.o:/home/lheck/.opam/4.09.0/.opam-switch/build/ocaml-base-compiler.4.09.0/runtime/backtrace.c:31: multiple definition of `caml_debug_info'; backtrace_byt_bpic.o:/home/lheck/.opam/4.09.0/.opam-switch/build/ocaml-base-compiler.4.09.0/runtime/backtrace_byt.c:47: first defined here
# collect2: error: ld returned 1 exit status
# make[1]: *** [Makefile:297: libcamlrun_shared.so] Error 1
# make[1]: Leaving directory '/home/lheck/.opam/4.09.0/.opam-switch/build/ocaml-base-compiler.4.09.0/runtime'
# make: *** [Makefile:345: coldstart] Error 2



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build ocaml-base-compiler 4.09.0
└─ 
┌─ The following changes have been performed (the rest was aborted)
│ ∗ install base-bigarray base
│ ∗ install base-threads  base
│ ∗ install base-unix     base
└─ 
# Run eval $(opam env) to update the current shell environment
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] Y
➜  plotkicadsch git:(master) 
```

--- a/packages/ocaml-base-compiler/ocaml-base-compiler.4.09.0/files/fix-gcc10.patch
+++ b/packages/ocaml-base-compiler/ocaml-base-compiler.4.09.0/files/fix-gcc10.patch
@@ -1,0 +1,34 @@
+commit 3f10a16153308f967149917585d2bc0b9c06492c
+Author: Anil Madhavapeddy <anil@recoil.org>
+Date:   Sun Jun 21 18:40:27 2020 +0100
+
+    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
+    
+    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+
+diff --git a/configure b/configure
+index 9a78a4554..0c54b560b 100755
+--- a/configure
++++ b/configure
+@@ -12412,7 +12412,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
+ -fno-builtin-memcmp";
+       internal_cflags="$gcc_warnings" ;; #(
+   gcc-*) :
+-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
++    common_cflags="-O2 -fno-strict-aliasing -fwrapv -fcommon";
+       internal_cflags="$gcc_warnings" ;; #(
+   msvc-*) :
+     common_cflags="-nologo -O2 -Gy- -MD"
+diff --git a/configure.ac b/configure.ac
+index f5d8a2687..775e0e2db 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -540,7 +540,7 @@ AS_CASE([$host],
+ -fno-builtin-memcmp";
+       internal_cflags="$gcc_warnings"],
+     [gcc-*],
+-      [common_cflags="-O2 -fno-strict-aliasing -fwrapv";
++      [common_cflags="-O2 -fno-strict-aliasing -fwrapv -fcommon";
+       internal_cflags="$gcc_warnings"],
+     [msvc-*],
+       [common_cflags="-nologo -O2 -Gy- -MD"

--- a/packages/ocaml-base-compiler/ocaml-base-compiler.4.09.0/opam
+++ b/packages/ocaml-base-compiler/ocaml-base-compiler.4.09.0/opam
@@ -30,7 +30,11 @@ url {
   src: "https://github.com/ocaml/ocaml/archive/4.09.0.tar.gz"
   checksum: "md5=76ac39570fc88b16fda2a94db7cd5cf3"
 }
-extra-files: ["ocaml-base-compiler.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+patches: ["fix-gcc10.patch"]
+extra-files: [
+  ["fix-gcc10.patch" "md5=f406119ae0091835cdf158d7d0ff53f7"]
+  ["ocaml-base-compiler.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+]
 post-messages: [
   "A failure in the middle of the build may be caused by build parallelism
    (enabled by default).

